woocommerce/woocommerce
# Add a new plugin to the monorepo for the docs project

### Changes proposed in this Pull Request:

This PR adds a WIP plugin that will be used to ingest Markdown content from remote sources and turn it into Wordpress posts.

I've opted to open a PR here at this stage in the process to keep the individual PRs as small and reviewable as possible, but I'm happy if its preferred to make the plugin more full featured before opening a PR. Let me know what you (the reviewer) think.

Due to preference in how I build plugins I've opted for this to be a single page application. The plugin renders just one admin page with a div. The div is where we mount a React app that's built and compiled with TS and Webpack.

The plan for this plugin is to store a list of manifests that can be used to source remote content on a schedule. I've added a REST API endpoint that allows fetching the manifest list for display in admin. To avoid creating new tables I've opted to use Wordpress options to store this data, I think we'll only have a handful of custom data we can store in options and for everything else this plugin does it will create Wordpress posts.

In follow-up PRs I'll add CRUD for the manifests and use action scheduler to schedule the fetching and parsing of manifests. The structure of a manifest is still TBD. 

This plugin is not built with turbo yet, and I'd rather avoid that until it does something useful.

Note that I'm intentionally building this plugin without any dependency on WooCommerce, you won't need WooCommerce for this plugin to function.

### How to test the changes in this Pull Request:

1. Setup a local dev env where you can mount this new plugin. 
2. In the project root run `pnpm i` if you haven't already
3. In the new plugin subdirectory run `pnpm build` this will do the webpack build
4. Now run your dev wordpress site, login to wp-admin, there should be a new top-level menu item for the plugin.
5. Navigate to the menu item, the page should load without errors, it should display a basic screen like the below screenshot:

<img width="884" alt="Screenshot 2023-05-23 at 11 20 19 AM" src="https://github.com/woocommerce/woocommerce/assets/1281828/4acf01d1-af59-462c-b39e-b004e562fe44">
